features/bsm/vaadin-adminpage/src/main/java/org/opennms/netmgt/bsm/vaadin/adminpage/BusinessServiceEdgeEditWindow.java

```diff
@@ -312,6 +318,37 @@ public BusinessServiceEdgeEditWindow(final BusinessService businessService,
 
             final int weight = Integer.parseInt(m_weightField.getValue());
 
+            /**
+             * in the case edge is not null, remove the old object...
+             */
+            if (edge != null) {
+                switch (edge.getType()) {
+                    case CHILD_SERVICE: {
+                        Set<ChildEdge> edges = businessService.getChildEdges();
+                        edges.remove(edge);
+                        businessService.setChildEdges(edges);
+                        break;
+                    }
+
+                    case IP_SERVICE: {
+                        Set<IpServiceEdge> edges = businessService.getIpServiceEdges();
+                        edges.remove(edge);
+                        businessService.setIpServiceEdges(edges);
+                        break;
+                    }
+
+                    case REDUCTION_KEY: {
+                        Set<ReductionKeyEdge> edges = businessService.getReductionKeyEdges();
+                        edges.remove(edge);
+                        businessService.setReductionKeyEdges(edges);
+                        break;
+                    }
+                }
+            }
```

Comment:
Can we move this to the BusinessServiceManager or BusinessService object? 
How about creating a `getEdges(EdgeType)` and `setEdges(EdgeType, Collection)` method?